### ISSUE TITLE
Provide support for Abbreviations or Acronyms

### ISSUE THREAD

**Issue by mgifford** [ckeditor/ckeditor-dev#2046]
## Web Accessibility

CKEditor should really support a `<abbr>` button.  This isn't a new issue. 

https://dev.ckeditor.com/ticket/1374

Feature request

## There should be a button that makes it easier to add a definition for an acronym

 I've linked to it from this Drupal issue where I raised it earlier ​https://www.drupal.org/node/2307549

This is part of the ATAG standard B3 ​http://www.w3.org/TR/ATAG20/#principle_b3

It should be easier to prompt users to insert abbreviations like: <abbr title="HyperText Markup Language">HTML</abbr>

by looking for groups of all caps. This shouldn't be mandatory, but just a reminder that this will help users who may not be familiar with the jargon.

I would suggest: UNCED, CIDA, NATO - would prompt a user for a definition. unced, cida, Nato - would not.

**Comment by wwalc** [comment]
The abbreviation plugin is actually presented in a tutorial:
https://docs.ckeditor.com/ckeditor4/latest/guide/plugin_sdk_sample_1.html

You may see it live here:
https://sdk.ckeditor.com/samples/abbr.html

Source code:
https://github.com/ckeditor/ckeditor-docs-samples/tree/master/tutorial-abbr-acf/abbr

**Comment by mgifford** [comment]
Ok.. Any reason not to build it in by default? Would be great to have folks actually use it!

**Comment by fuzzbomb** [comment]
`<acronym>` was removed as of HTML5

**Comment by mgifford** [comment]
I had used the term acronym when describing the problem, but the links that @wwalc provided give you 
`<abbr title="What You See Is What You Get">WYSIWYG</abbr>`

**Comment by fuzzbomb** [comment]
Yeah that's good output from the SDK sample.  Title attributes on ABBR tags are partially "accessibility supported" (a WCAG term).

Good: Several screen readers make ABBR title attributes available.

Bad: ABBR title attributes are not available to sighted keyboard-only users. Sighted mouse users get the title attribute on hover.  Not sure about speech control or switch access users.

**Comment by mgifford** [comment]
"ABBR title attributes are not available to sighted keyboard-only users." - This sounds like a browser issue though.

**Comment by fuzzbomb** [comment]
> Ok.. Any reason not to build it in by default? Would be great to have folks actually use it!

A big +100 to that.

ABBR is an element which survived in HTML5, and has good screen reader accessibility.

Having the ABBR button in the standard CKEditor build would be a great help for CMS (etc.) aiming to support:

WCAG 2.0 - success criterion 3.1.4 Abbreviations 
ATAG 2.0 - guideline Principle B.2: Authors are supported in producing accessible content

**Comment by fuzzbomb** [comment]
Yeah, it's the general problem of user-agents treating title attributes as a tooltip, but only for hover.  Same for abbr as most other elements.

**Comment by mgifford** [comment]
How do we postpone this until there is better browser support?

**Comment by iparr** [comment]
Browser support still looks poor.

However, supporting `<abbr title="">` does represent a net gain in many cases for helping users understand content. Any love for this?

(Note: I'm using _Craft CMS v5_ and reliably adding plugins to CKEditor is a bit of a chore)
